src/mlir/driver.cc

```diff
@@ -106,4 +105,73 @@ namespace mlir::verona
     module->print(out);
     return llvm::Error::success();
   }
+
+  llvm::Error Driver::emitLLVM(llvm::StringRef filename)
+  {
+    if (filename.empty())
+      return runtimeError("No output filename provided");
+
+    // Required passes to shape the IR into an acceptable LLVM format
+    passManager.addPass(mlir::createVeronaManglingPass());
+    // The lowering "pass manager"
+    passManager.addPass(mlir::createLowerToLLVMPass());
+
+    // First lower to LLVM dialect
+    if (mlir::failed(passManager.run(module.get())))
+    {
+      module->dump();
+      return runtimeError("Failed to lower to LLVM dialect");
+    }
+
+    // Register the translation to LLVM IR with the MLIR context.
+    mlir::registerLLVMDialectTranslation(*module->getContext());
+
+    // Then lower to LLVM IR
+    llvm::LLVMContext llvmContext;
+    auto llvm = mlir::translateModuleToLLVMIR(module.get(), llvmContext);
+    if (!llvm)
+      return runtimeError("Failed to lower to LLVM IR");
+
+    // Optimise if requested
+    if (optLevel)
+    {
+      // Initialize LLVM targets.
+      llvm::InitializeNativeTarget();
```

Comment:
It would be good to thread a target triple through this early so we don't build something without cross-compilation support and have to add it later.  I presume the Verona lowering thing will eventually be parameterised on some aspects of the target, but most of the Verona MLIR will be target-independent.

Reply:
Adding target triple to this would be fairly trivial at any point, at least until we start doing target-specific things on our end (foreign code, struct layout, etc). Right now I'm focusing on target agnostic parts on purpose, to get the overall structure of the compiler in shape.

When we add target-specific things, the triple will be an obvious and trivial change, and I would prefer to defer it to when we do that, to make sure we're adding all the right hooks at the right places. If I add it now, I won't test and could be as bogus as not having it.

I'll reply to your review on the document in separate, as a single review.